app/main.cpp

```diff
@@ -369,5 +379,9 @@ int main(int argc, char *argv[])
 
     int err = app.exec();
 
+    // Give worker tasks time to properly exit. Fixes PendingQuitTask
+    // sometimes freezing and blocking process exit.
+    QThreadPool::globalInstance()->waitForDone(30000);
```

Comment:
This is interesting. Can you tell me more about what you were seeing here?

I think we do have some race conditions on process exit as I've seen a few random crashes locally deep inside Qt frameworks. I'm somewhat tempted to `exit(err)` at the end of this function to avoid the pointless crashes, but it would be nice to know what is actually going on at that time (since ComputerManager should stop polling in its destructor).

Reply:
I can fairly easily reproduce this with debug build without debugger. After ending the session moonlight.exe's process is left running even though its window is already closed. Attaching to the process shows these three threads which have Moonlight's sources visible in call stack:
![main-thread](https://user-images.githubusercontent.com/1950698/49341195-cce34e00-f652-11e8-8340-341635517000.png)
![pending-quit-task](https://user-images.githubusercontent.com/1950698/49341201-dbca0080-f652-11e8-8122-e463646ed132.png)
![update-app-list](https://user-images.githubusercontent.com/1950698/49341202-df5d8780-f652-11e8-8151-029fbb7090b4.png)
Unfortunately I don't have matching pdb files for Qt's binaries, so can't see where exactly the main thread is hanging at.